source/Calamari/Integration/Scripting/WindowsPowerShell/PowerShellBootstrapper.cs

```diff
@@ -195,21 +188,27 @@ static void WriteScriptModules(VariableDictionary variables, string parentDirect
             }
         }
 
-        static void WriteVariableDictionary(CalamariVariableDictionary variables, StringBuilder output)
+        static (string encrypted, string iv) GetEncryptedVariablesString(CalamariVariableDictionary variables)
         {
-            output.AppendLine("$OctopusParameters = New-Object 'System.Collections.Generic.Dictionary[String,String]' (,[System.StringComparer]::OrdinalIgnoreCase)");
+            var sb = new StringBuilder();
             foreach (var variableName in variables.GetNames().Where(name => !SpecialVariables.IsLibraryScriptModule(name)))
             {
-                var variableValue = variables.IsSensitive(variableName)
-                    ? EncryptVariable(variables.Get(variableName))
-                    : EncodeValue(variables.Get(variableName));
-
-                output.Append("$OctopusParameters[").Append(EncodeValue(variableName)).Append("] = ").AppendLine(variableValue);
+                var value = variables.Get(variableName);
+                var encryptedValue = value == null ? "nul" : EncodeAsBase64(value); // "nul" is not a valid Base64 string
```

Comment:
It would be nice to have this comment up here too: https://github.com/OctopusDeploy/Calamari/pull/276/files#diff-622d56f3b1f1e29f968b8da02b798fe4R226

I got confused when I saw `-eq 'nul'`